roles/repositories_sync/tasks/main.yml

```diff
@@ -0,0 +1,29 @@
+---
+- name: Sync repository
+  theforeman.foreman.repository_sync:
+    username: "{{ username | default(omit) }}"
+    password: "{{ password | default(omit) }}"
+    server_url: "{{ server_url | default(omit) }}"
+    validate_certs: "{{ validate_certs | default(omit) }}"
+    organization: "{{ organization }}"
+    product: "{{ item.0.name }}"
+    repository: "{{ item.1.name }}"
```

Comment:
If you omit `repository` the module will sync the whole product, which in most cases is what the user wants to do anyways.

Reply:
If we go for the batching, we might want to iterate over the repositories to even out the load better.
If we do one product at a time, this might just be the easiest approach to batching.

Just my two thoughts here...

Reply:
Yeah, agree, "by product" is probably the cheapest batching we can get

Reply:
OK. Will update to just sync by product and not async. Can an issue be opened to better queue requests in foreman? At the end of the day, it really shouldn't be the automation's responsibility to batch requests and manage load for the system.